Lib/pathlib/__init__.py

```diff
@@ -587,9 +587,13 @@ def iterdir(self):
     def _scandir(self):
         return os.scandir(self)
 
+    def _entry_str(self, entry):
```

Comment:
Can this be `_direntry_str`? Or maybe `_direntry_name` if that's what it's doing?

I can't figure out from the code why the function works - the dependency on `str(self)` feels out of place. So maybe explaining why our own value matters would help?

Reply:
I've added a comment to hopefully explain. `PathBase._scandir()` yields `PathBase` objects, whereas `Path._scandir()` yields `os.DirEntry` objects. The implementation of `glob()` uses intersection of their APIs, with a little bit of help from `_make_child_direntry()` and `_direntry_str()` to smooth over the differences.